src/wallet/wallet.cpp

```diff
@@ -4315,6 +4254,66 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
     return walletInstance;
 }
 
+bool CWallet::UpgradeWallet(int version, std::string& out)
+{
+    int prev_version = GetVersion();
+    int nMaxVersion = version;
+    if (nMaxVersion == 0) // the -upgradewallet without argument case
+    {
+        WalletLogPrintf("Performing wallet upgrade to %i\n", FEATURE_LATEST);
+        nMaxVersion = FEATURE_LATEST;
+        SetMinVersion(FEATURE_LATEST); // permanently upgrade the wallet immediately
+    }
+    else
+        WalletLogPrintf("Allowing wallet upgrade up to %i\n", nMaxVersion);
+    if (nMaxVersion < GetVersion())
+    {
+        out = _("Cannot downgrade wallet");
+        return false;
+    }
+    SetMaxVersion(nMaxVersion);
+
+    // Upgrade to HD if explicit upgrade
+    LOCK(cs_wallet);
```

Comment:
This comment isnt related to the lock, is it ?

Reply:
Removed